Server/mods/deathmatch/logic/packets/CEntityAddPacket.cpp

```diff
@@ -297,6 +297,9 @@ bool CEntityAddPacket::Write(NetBitStreamInterface& BitStream) const
                     health.data.fValue = pObject->GetHealth();
                     BitStream.Write(&health);
 
+                    // is object break?
+                    BitStream.WriteBit(pObject->GetHealth() <= 0);
```

Comment:
Again: you have to check if the bitstream supports this.

Reply:
I understand client-side checking because the server may not send this bit, but why server-side checking? If the server is older, it won't have this piece of code, right?

Reply:
If an older client doesn't support this, and the newer server sends this bit, the client will read garbage.

Reply:
Yes, I know on the client side why we check. But why do we check this with WriteBit? If the server is older, it simply does not have this line of code

Reply:
New server breaks old clients with unexpected data in stream

Reply:
So checking BitStream::Can also checks whether the client (who will receive the packet) has the same version as the server? Because I still don't understand why we check on the server side whether the bit can be sent, when the older server simply won't send it because it doesn't have this piece of code

Reply:
@botder ?

Reply:
> So checking BitStream::Can also checks whether the client (who will receive the packet) has the same version as the server?

So checking BitStream::Can also checks whether the client (who will receive the packet) has the same version as the server?

Reply:
I don't know what the server does - I would have to look it up - I assume that it creates a BitStream for every player and then uses the Packet::Write method for every instance.

Reply:
Okey, done.